core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -4164,16 +4164,13 @@ class ReplicaManagerTest {
         mock(classOf[FetchDataInfo])
       }).when(spyRLM).read(any())
 
-      // Get the current type=DelayedRemoteFetchMetrics,name=ExpiresPerSec metric value before fetching
-      val curExpiresPerSec = safeYammerMetricValue("type=DelayedRemoteFetchMetrics,name=ExpiresPerSec").asInstanceOf[Long]
+      val curExpiresPerSec = DelayedRemoteFetchMetrics.expiredRequestMeter.count()
       replicaManager.fetchMessages(params, Seq(tidp0 -> new PartitionData(topicId, fetchOffset, 0, 100000, Optional.of[Integer](leaderEpoch), Optional.of[Integer](leaderEpoch))), UnboundedQuota, fetchCallback)
       // advancing the clock to expire the delayed remote fetch
       timer.advanceClock(2000L)
 
-      // verify the metric value is incremented since the delayed remote fetch is expired
-      TestUtils.waitUntilTrue(() => curExpiresPerSec + 1 == safeYammerMetricValue("type=DelayedRemoteFetchMetrics,name=ExpiresPerSec").asInstanceOf[Long],
-        "The ExpiresPerSec value is not incremented. Current value is: " +
-          safeYammerMetricValue("type=DelayedRemoteFetchMetrics,name=ExpiresPerSec").asInstanceOf[Long])
+      // verify the DelayedRemoteFetchMetrics.expiredRequestMeter.mark is called since the delayed remote fetch is expired
+      assertEquals(curExpiresPerSec + 1, DelayedRemoteFetchMetrics.expiredRequestMeter.count())
```

Comment:
I think we should use use `TestUtils.waitUntilTrue` here because the `DelayedRemoteFetchMetrics` is marked in a separate thread, we can't make sure it will be triggered immediately.

Reply:
ah i see. thanks for the comment @showuon ! 
let me address that

Reply:
Looks like it didn't appear in the rebuild. Let me re-trigger the CI again.
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-15463/13/

Reply:
OK, the testRemoteFetchExpiresPerSecMetric are all passed. Please use `TestUtils.waitUntilTrue` here, otherwise, LGTM! Thanks.

Reply:
@showuon just updated, thanks for the comment!